src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -74,73 +51,21 @@ struct combine_latest_observer_strategy
         const auto observer = disposable->get_observer_under_lock();
         disposable->get_values().template get<I>().emplace(std::forward<T>(v));
 
-        disposable->get_values().apply([this, &observer](const std::optional<Args>&... vals) {
-            if ((vals.has_value() && ...))
-                observer->on_next(disposable->get_selector()(vals.value()...));
-        });
-    }
-
-    void on_error(const std::exception_ptr& err) const
-    {
-        disposable->dispose();
-        disposable->get_observer_under_lock()->on_error(err);
+        disposable->get_values().apply(&apply_impl<decltype(disposable)>, disposable, observer);
     }
 
-    void on_completed() const
+private:
+    template<typename TDisposable>
+    static void apply_impl(const TDisposable& disposable, const pointer_under_lock<Observer>& observer, const std::optional<Args>&... vals)
     {
-        if (disposable->decrement_on_completed())
-        {
-            disposable->dispose();
-            disposable->get_observer_under_lock()->on_completed();
-        }
+        if ((vals.has_value() && ...))
+            observer->on_next(disposable->get_selector()(vals.value()...));
     }
 };
 
 template<typename TSelector, rpp::constraint::observable... TObservables>
-struct combine_latest_t
+struct combine_latest_t final : public combining_operator_t<combine_latest_disposable, combine_latest_observer_strategy, TSelector, TObservables...>
```

Comment:
using?

Reply:
you mean to remove final?

Reply:
no, i mean we don't need struct in this case, just using to combining_operator_t...